cmd/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+// current implementation is for sqlite
+import (
+	"net/http"
+
+	gin "github.com/gin-gonic/gin"
+
+	sqlite_controller "github.com/leechongyan/Studtor_backend/internal/controllers/sqlitecontroller"
+)
+
+func main() {
+	// Set the router as the default one shipped with Gin
+	router := gin.Default()
+
+	// Serve 404 page when page not found
+	router.NoRoute(func(c *gin.Context) {
+		c.JSON(404, gin.H{"code": "PAGE_NOT_FOUND", "message": "Page not found"})
+	})
+
+	// "/ping" - example GET request endpoint
+	router.GET("/ping", func(c *gin.Context) {
+		c.JSON(200, gin.H{
+			"message": "pong",
+		})
+	})
+
+	// "/students/:id" - endpoint to retrieve student from db by ID
+	router.GET("/students/:id", func(c *gin.Context) {
+		id := c.Param("id")
+		name, err := sqlite_controller.SelectStudentByID(id)
+		if err != nil {
+			c.String(http.StatusBadRequest, "StatusBadRequest")
```

Comment:
Would it be better to define a function to return us the appropriate error string based on the type of error? This would provide more information to the user regarding the actual error!

Reply:
Resolved in PR #19